src/data/finsupp/basic.lean

```diff
@@ -595,12 +595,20 @@ lemma prod_ite_eq (f : α →₀ M) (a : α) (b : α → M → N) :
   f.prod (λ x v, ite (a = x) (b x v) 1) = ite (a ∈ f.support) (b a (f a)) 1 :=
 by { dsimp [finsupp.prod], rw f.support.prod_ite_eq, }
 
+@[simp] lemma sum_ite_self_eq {N : Type*} [add_comm_monoid N] (f : α →₀ N) (a : α) :
+  f.sum (λ x v, ite (a = x) v 0) = f a :=
```

Comment:
`ite (a = x) v 0` is basically `finsupp.single x v a`, for which there are plenty of lemmas about. Does the ite appear in your proof in a place where finsupp.single could be used instead?

Reply:
Although this lemma is probably good to have anyway

Reply:
I didn't succeed in massaging the use (it's in `orthonormal.inner_right_finsupp`) to this form.